integration/benchmarks_test.go

```diff
@@ -55,3 +55,31 @@ func BenchmarkQuery(b *testing.B) {
 		})
 	}
 }
+
+func BenchmarkReplaceLargeDocument(b *testing.B) {
+	provider := shareddata.LargeDocumentBenchmarkValues
+
+	s := setup.SetupWithOpts(b, &setup.SetupOpts{
+		BenchmarkProvider: &provider,
+	})
+	ctx, coll := s.Ctx, s.Collection
+
+	filter := bson.D{{"_id", 0}}
+	runsCount := 1
+
+	b.Run(provider.Hash(), func(b *testing.B) {
+		for i := 0; i < b.N; i++ {
+			var doc bson.D
+			err := coll.FindOne(ctx, filter).Decode(&doc)
+			require.NoError(b, err)
+
+			doc[runsCount].Value = i * 11111
```

Comment:
Not quite sure what is `runsCount`. It will always be 1 right? Because `runsCount++` is outside for loop.

What is the significance of `doc["1"].Value`? :thinking: Is it the first non _id field value?

Reply:
> Not quite sure what is `runsCount`. It will always be 1 right?

testing.Benchmark will run the benchmark multiple times, each time with increased b.N number based on the execution time. We count those runs and use them to update next and next fields inside the document, that's because we want to make sure that we actually replace the data (to satisfy `require.Equal(b, int64(1), updateRes.ModifiedCount)`)

Reply:
I see, you are using that to generate random value to modify the document.